workspaces/ballerina/ballerina-extension/src/features/ai/service/libs/langlibs.ts

```diff
@@ -14,7 +14,240 @@
+**reduce(arr, func, initial)**: Combine elements to single value.
+**indexOf(arr, val, start?)**: First occurrence index or ().
+**push(arr, ...vals)**: Add to end. Modifies in place.
+**pop(arr)**: Remove and return last. Panics if empty.
+**shift(arr)**: Remove and return first. Panics if empty.
+**unshift(arr, ...vals)**: Add to start.
+**sort(arr, dir?, key?)**: Sort elements. dir: "ascending"|"descending". Returns new array.
+**fromBase64(str)**: Decode Base64 to byte[].
+
+## lang.string
+
+**length(str)**: Code point count.
+**join(sep, ...strs)**: Join with separator.
+**codePointCompare(s1, s2)**: Lexicographic compare. Returns int (<0, 0, >0).
+**concat(...strs)**: Join strings.
+**endsWith(str, sub)**: Check suffix. Returns boolean.
+**fromBytes(bytes)**: UTF-8 bytes to string.
+**fromCodePointInt(cp)**: Code point to Char.
+**fromCodePointInts(cps)**: Code points to string.
+**getCodePoint(str, i)**: Get code point at index.
+**includes(str, sub, start?)**: Contains check. Returns boolean.
+**indexOf(str, sub, start?)**: Find substring. Returns index or ().
+**matches(str, re)**: Full regex match. Returns boolean.
+**startsWith(str, sub)**: Check prefix. Returns boolean.
+**substring(str, start, end?)**: Extract substring.
+**toBytes(str)**: String to UTF-8 bytes.
+**toCodePointInt(ch)**: Char to code point.
+**toCodePointInts(str)**: String to code point array.
+**toLowerAscii(str)** / **toUpperAscii(str)**: ASCII case conversion.
+**trim(str)**: Remove ASCII whitespace from ends.
+
+## lang.map
+
+**length(map)**: Entry count.
+**get(map, key)**: Get value. Panics if missing.
+**hasKey(map, key)**: Key exists check. Returns boolean.
+**keys(map)**: All keys as string[].
+**entries(map)**: [key, value] pairs as map.
+**toArray(map)**: All values as array.
+**removeAll(map)**: Clear all entries.
+**removeIfHasKey(map, key)**: Safe remove. Returns value or ().
+
+## lang.int
+
+**fromString(str)**: Parse decimal string. No hex, allows +/-.
+**toHexString(n)**: int to hex (lowercase, no 0x).
+**fromHexString(str)**: Parse hex. Allows +/-, no 0x.
+
+## lang.float
+
+**fromString(str)**: Parse float. Allows NaN, Infinity, +/-.
+**exp(x)**: e^x. Use for exponential/growth calculations.
+**toHexString(x)**: Float to hex string (IEEE 754).
+**fromHexString(str)**: Parse hex float.
+
+## lang.decimal
+
+**fromString(str)**: Parse decimal string. Use for precision/financial.
+**max(n, ...ns)** / **min(n, ...ns)**: Find extreme values.
+**sum(...ns)**: Add decimals.
+
+## lang.boolean
+
+**fromString(str)**: Parse "true"|"false" (case-sensitive).
+
+## lang.error
+
+**message(e)**: Get error message string.
+**cause(e)**: Get chained error or ().
+**detail(e)**: Get readonly detail record.
+
+## lang.regexp
+
+**IMPORTANT: Regex Syntax Rules**
+- Regex literals use backtick syntax: "re \`pattern\`", Ex: re \`[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}\`
+- The closing backtick MUST appear before any method call
```

Comment:
Idealy, this line can be remove after adding samples in L160, Lets test it and see